### PR TITLE
Homework 3: completed

### PR DESCRIPTION
**Pull request by ValentinBodrov**

### CODE REVIEW

File: src/main/java/hw3/ex1/HomePage.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    @FindBy(id = "password")
+    private WebElement passwordTextField;
+
+    @FindBy(id = "login-button")
+    private WebElement loginButton;
+
+    @FindBy(id = "user-name")
+    private WebElement usernameLabel;
+
+    @FindBy(xpath = "//ul[@class='uui-navigation nav navbar-nav m-l8']")
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: There is not necessarily has all list of the classes in the locator. `//ul[@class='uui-navigation nav navbar-nav m-l8']` please choose one which more prefarable

File: src/main/java/hw3/ex1/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    private WebElement header;
+
+    @FindBy(xpath = "//div[@class='row clerafix benefits']")
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: There is not necessarily has all list of the classes in the locator. `//ul[@class='uui-navigation nav navbar-nav m-l8']` please choose one which more prefarable

File: src/main/java/hw3/ex1/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    private WebElement benefits;
+
+    @FindBy(xpath = "//h3[@name='main-title']")
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: Could be used DOM locator

File: src/main/java/hw3/ex1/HomePage.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    private WebElement mainHeader;
+
+    @FindBy(xpath = "//p[@name='jdi-text']")
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: Could be used DOM locator

File: src/main/java/hw3/ex1/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    @FindBy(id = "iframe")
+    private WebElement iframe;
+
+    @FindBy(id = "epam_logo")
+    private WebElement epamLogo;
+
+    @FindBy(linkText = "JDI GITHUB")
+    private WebElement linkElement;
+
+    @FindBy(xpath = "//div[@name='navigation-sidebar']")
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: Could be used DOM locator

File: src/main/java/hw3/ex1/HomePage.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    private WebElement leftSection;
+
+    @FindBy(tagName = "footer")
+    private WebElement footer;
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: I think, that footer we have on the all pages. Not only on the `Home`

File: src/main/java/hw3/ex1/HomePage.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+    public String getUsernameLabelText() {
+        return usernameLabel.getText();
+    }
+
+    public ArrayList<String> getHeaderItems() {
+        List<WebElement> headerItems = header.findElements(By.tagName("li"));
+        ArrayList<String> actualHeaderTexts = new ArrayList<>();
+        for (WebElement headerItem : headerItems) {
+            if (headerItem.getText().equals("")) {
```
Comment (khda91) [inline src/main/java/hw3/ex1/HomePage.java]: It si better and safer has next structure for the comparing `"".equals(headerItem.getText())`

File: src/main/java/hw3/ex2/components/HeaderMenu.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import org.openqa.selenium.support.PageFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class HeaderMenu {
+
+    private WebDriver driver;
+
+    @FindBy(xpath = "//ul[@class='uui-navigation nav navbar-nav m-l8']")
```
Comment (khda91) [inline src/main/java/hw3/ex2/components/HeaderMenu.java]: There is not necessarily has all list of the classes in the locator. `//ul[@class='uui-navigation nav navbar-nav m-l8']` please choose one which more prefarable

**Review by khda91 — COMMENTED** [review]
Please resolve comments

File: src/main/java/hw3/ex1/HomePage.java (hunk 1 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: Solved: now I'm using only `navbar-nav` class. I also changed the locator-type: from *xpath* to *classname*

File: src/main/java/hw3/ex1/HomePage.java (hunk 2 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: Solved. If we're speaking about line 34, I changed that like in previous point, now I'm using only `benefits` class

File: src/main/java/hw3/ex1/HomePage.java (hunk 3 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: Solved: I change it on `@FindBy(name = "main-title")`

File: src/main/java/hw3/ex1/HomePage.java (hunk 4 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: Solved: I change it on `@FindBy(name = "jdi-text")`

File: src/main/java/hw3/ex1/HomePage.java (hunk 5 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: Solved: I change it on `@FindBy(name = "navigation-sidebar")`

File: src/main/java/hw3/ex1/HomePage.java (hunk 6 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: Solved: I carried it over in the *AbstractBasePage* with its getter-method

File: src/main/java/hw3/ex1/HomePage.java (hunk 7 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex1/HomePage.java]: I fixed it, thanks for the suggestion

File: src/main/java/hw3/ex2/components/HeaderMenu.java (hunk 8 above)
Comment (ValentinBodrov) [reply src/main/java/hw3/ex2/components/HeaderMenu.java]: Solved: now I'm using `@FindBy(className = "navbar-nav")`

**Review by khda91 — APPROVED** [review]
Approved. Could be merged to the master branch

**Pull request merged**